src/org/zaproxy/zap/spider/Spider.java

```diff
@@ -172,13 +172,13 @@ public Spider(String id, ExtensionSpider extension, SpiderParam spiderParam, Con
 		this.id = id;
 		this.spiderParam = spiderParam;
 		this.connectionParam = connectionParam;
-		this.model = model;
+		this.model = model;		
+		this.extension = extension;		
```

Comment:
Does this really need to be moved from [L181](https://github.com/zaproxy/zaproxy/pull/3028/commits/00b4690e76aab56f86329d7574382ede7fbde703#diff-1f3205859ebf47606f09a0c795150206L181) to here?